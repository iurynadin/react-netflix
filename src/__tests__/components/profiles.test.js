import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
    it('renders <Profiles /> with populoated data', () => {
        const { container, getByTestId, getByText } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}} >
                        <Profiles.Picture src='/images/karl.png' data-testid="profile-picture" />
                        <Profiles.Name>Iury Nadin</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        )

        expect(getByText("Who's watching?")).toBeTruthy();
        expect(getByTestId("profile-picture")).toBeTruthy();
        expect(getByText("Iury Nadin")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders <Profiles /> with populoated data but misc profile image', () => {
        const { container, getByTestId, getByText } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}} >
                        <Profiles.Picture data-testid="profile-picture-misc" />
                        <Profiles.Name>Iury Nadin</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        )

        expect(getByText("Who's watching?")).toBeTruthy();
        expect(getByTestId("profile-picture-misc")).toBeTruthy();
        expect(getByText("Iury Nadin")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});