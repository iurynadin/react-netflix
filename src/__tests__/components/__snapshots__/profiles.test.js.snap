// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders <Profiles /> with populoated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 lfIDZX"
>
  <h1
    class="profiles__Title-bxtkkv-1 kjBRZT"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 iCNnxQ"
  >
    <div
      class="profiles__User-bxtkkv-6"
    >
      <img
        class="profiles__Picture-bxtkkv-4 gqRDnr"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 CaEMo"
      >
        Iury Nadin
      </p>
    </div>
  </ul>
</div>
`;

exports[`<Profiles /> renders <Profiles /> with populoated data but misc profile image 1`] = `
<div
  class="profiles__Container-bxtkkv-0 lfIDZX"
>
  <h1
    class="profiles__Title-bxtkkv-1 kjBRZT"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 iCNnxQ"
  >
    <div
      class="profiles__User-bxtkkv-6"
    >
      <img
        class="profiles__Picture-bxtkkv-4 gqRDnr"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-bxtkkv-3 CaEMo"
      >
        Iury Nadin
      </p>
    </div>
  </ul>
</div>
`;
